### PR TITLE
*: optimize status definition

### LINKED ISSUE #5399: server/schedule/plan: should we classify the status code?
## Enhancement Task
In server/schedule/plan, we list the status code to indicate the failed reasons when try to schedule or check as bellow:
[https://github.com/tikv/pd/blob/99e38eb7b4454cc283c0fea99518753a196b9b50/server/schedule/plan/status.go#L22-L66](https://github.com/tikv/pd/blob/99e38eb7b4454cc283c0fea99518753a196b9b50/server/schedule/plan/status.go#L22-L66)

Here I think we could classify the status like:
- status for store
    - reason1
    - ..
- status for region
   - status for placement rule
   - ..
- other status

which can help use to identify the kind of failed-reason faster, meanwhile, it will be easier to add a new status(reason), the new status code could always be together with the kind it belongs.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5399.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/schedule/plan/status.go [hunk 1]
```diff
@@ -14,83 +14,119 @@
...
-	// StatusStoreThrottled represents the store cannot be selected due to the limitation.
-	StatusStoreThrottled
-	// StatusStoreUnavailable represents the store cannot be selected due to it's state.
-	StatusStoreUnavailable
+)
+
+// normal status in most of situations.
+const (
+	// StatusScoreAlmostSame represents the plan is no need to be scheduled due to the score is almost the same.
+	StatusScoreAlmostSame = iota + 100
```
Comment (CabinfeverB) [inline server/schedule/plan/status.go:29]: which status should we use when the score of store1 is lower than store2

File: server/schedule/plan/status.go [hunk 2]
```diff
@@ -14,83 +14,119 @@
...
-	// StatusIsolationNotMatch represents the isolation cannot satisfy the requirement.
-	StatusIsolationNotMatch
-
-	// TODO: The below status is not used for now. Once it is used, Please remove this comment.
-
-	// StatusStoreNotExisted represents the store cannot be found in PD.
-	StatusStoreNotExisted
-	// StatusNoTargetRegion represents the target region of merge operation cannot be found.
+	// StatusRegionNotMatchRule represents the region does not match rule constraint.
+	StatusRegionNotMatchRule
```
Comment (CabinfeverB) [inline server/schedule/plan/status.go:88]: no statusText for StatusRegionNotMatchRule

**Review by CabinfeverB — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**